pymc/sampling/jax.py

```diff
@@ -532,15 +532,17 @@ def sample_jax_nuts(
 
     model = modelcontext(model)
 
-    if var_names is None:
-        var_names = model.unobserved_value_vars
+    if var_names is not None:
+        vars_ = [v for v in model.unobserved_value_vars if v.name in var_names]
```

Comment:
A more readable name? I guess mypy didn't like the redefinition?
```suggestion
        filtered_var_names = [v for v in model.unobserved_value_vars if v.name in var_names]
```